File: Python/car_fuel_system.py
class CarFuelSystem:
    def __init__(self):
        self.levelOfFuel = 0
        self.isStarted = False
        self.fuelEfficiency = 16

    def start_car(self):
        if self.isStarted == True:
            raise ValueError("Car is already started")
        else:
            self.isStarted = True

    def stop_car(self):
        if self.isStarted == False:
            raise ValueError("Car is already stopped")
        else:
            self.isStarted = False

    def get_level_of_fuel(self):
        return self.levelOfFuel

    def get_car_starting_status(self):
        return self.isStarted

    def refuel(self, litre):
        if litre <= 0:
            raise ValueError("Litres efficiency cannot be less than or equal to 0")
        elif litre + self.levelOfFuel > 50:
            raise ValueError("Highest fuel level is 50")
        else:
            self.levelOfFuel += litre


    def calcuate_fuel_consumed(self, distance):
        fuel_consumed = distance / self.fuelEfficiency
        result = f"{fuel_consumed:.2f} "
        return int(result)

    def drive_car(self, distance):
        if self.isStarted == False:
            raise ValueError("Car cannot be driven when it is not started")
        elif distance <= 0:
            raise ValueError("Distance is cannot be less than or equal to 0")
        else:
            fuel_consumed = self.calcuate_fuel_consumed(distance)
            self.levelOfFuel -= fuel_consumed

File: Java Assignment/CarFuelSystem.java
public class CarFuelSystem{
	private double levelOfFuel;
	private boolean isStarted ;
	private static final int fuelEfficiency = 16;	



	public void StartCar(){
		if (isStarted == true) throw new IllegalArgumentException("Car is started already");

		else
		 isStarted = true;
				
	}
	

	public void stopCar(){
		if (isStarted == false) throw new IllegalArgumentException("You cannot stop car when it is not started");
		else isStarted = false;

	}

	public double getLevlOfFuel(){
		return levelOfFuel;

	}


	public boolean getCarStartingStatus(){
		return isStarted;
		
	}

	public void refuel(double liters){
		if(liters <= 0) throw new IllegalArgumentException("liters cannot be less than or equal to zero");
		else if(liters + levelOfFuel > 50) throw new IllegalArgumentException("Highest fuel level is 50liters");
		else levelOfFuel += liters;
	
	}
	private double calCulateFuelConsumed(double distance){
		double fuelConsumed = distance/ fuelEfficiency;
		String formatted = String.format("%.2d", fuelConsumed);
		int formattedInt = Integer.parseInt(formatted);
		return formattedInt;
		

	}
	public void drive(int distance){
		if(isStarted == false) throw new IllegalArgumentException("Car cannot be driven when it is not started");

		else if(distance <= 0) throw new IllegalArgumentException("You cannot go a negative distance");
		else if(levelOfFuel <= 0)throw new IllegalArgumentException("Car cannot be driven without fuel");
		else {
			double fuelConsumed =	calCulateFuelConsumed(distance);
			if(levelOfFuel - fuelConsumed < 0) levelOfFuel = 0;
			else{
				levelOfFuel -= fuelConsumed;

			}
		 }

	

	}


	


}